query/lexer.go

```diff
@@ -58,18 +71,19 @@ func tokenize(query string) ([]token, error) {
 			continue
 		}
 
-		if len(split) != 2 {
-			return nil, fmt.Errorf("can't tokenize \"%s\"", field)
-		}
-
 		if len(split[0]) == 0 {
 			return nil, fmt.Errorf("can't tokenize \"%s\": empty qualifier", field)
 		}
 		if len(split[1]) == 0 {
 			return nil, fmt.Errorf("empty value for qualifier \"%s\"", split[0])
 		}
 
-		tokens = append(tokens, newTokenKV(split[0], removeQuote(split[1])))
+		if len(split) == 2 {
+			tokens = append(tokens, newTokenKV(split[0], removeQuote(split[1])))
+		} else {
+			rest := strings.Join(split[2:], ":")
```

Comment:
I'm not sure this should really accept more than two `:` in a token.

Reply:
Do you have an alternative syntax in mind? I currently propose `metadata:github-url:https://github.com/author/myproject/issues/42`, but that means there will be 3 colons, even if the third is not interesting. I.e. if you ban > 2 colons, then colons are not allowed in metadata, which means URLs are not allowed in metadata, which would be sad.

Once we settle down on the syntax, I can take a look at the tests. :-) Thanks.

Reply:
Hu, good point. Man, those formal grammar lessons that I barely listen to are quite far now ;)

After some idea shuffling, I'm thinking that the good solution might be to use quotes. It's already possible to do that: `author:"René Descartes"`. So in your case it would be `metadata:github-url:"https://github.com/author/myproject/issues/42"`.

Also, the command line parse behave in a similar way, no? Would `git-bug ls --metadata github-url="https://github.com/author/myproject/issues/42"` be accepted? Not that it would be required as long as the shell is happy, but some uniformity would be nice.

What do you think?

Reply:
I like your proposal, and it may work for the webui, but the trouble is that the shell eats these quotes. So if I say meta:gh:"foo", I only get meta:gh:foo in go. You may require the user to type meta:gh:\"foo\", but that would be terrible usability.

Considering these, I would argue that either the current syntax is not that bad after all, or if you really would like some different syntax, then perhaps metadata:github-url=..., so it would be more obvious that the remaining characters are a value.

Which one would do you prefer?

Reply:
> So if I say meta:gh:"foo", I only get meta:gh:foo in go.

Is that really the shell or [cobra](https://github.com/spf13/cobra)? What do you get with `meta:gh:"foo bar"`?

--> I tested that, it's indeed go, **but**, go only unquote that and we get the args separated properly.

--> Actually, I had this problem before and to deal with it the `ls` command put the quotes back: https://github.com/MichaelMure/git-bug/blob/master/commands/ls.go#L93-L98

So, some shenanigan happen in the process but from the CLI side and parser side, it looks and behave as we expect it to.

> then perhaps metadata:github-url=...

That could be an alternative way. The grammar would be only 1-tuples (for the full text search) and 2-tuples for the filters/sorting. No 3-tuples.

In the later case, we would still need the quotes IMHO, to make the grammar a bit more regular. Having `:` suddenly not being a separator because we already found two doesn't sounds like a great idea. What if we have 4 parameters next? It's unlikely but it would break the previously valid queries.

Considering this, this later syntax doesn't have much interest, no? What do you think?

Reply:
Right. :-) One more thing I forgot is that the cmdline can always use `ls --metadata github-url=https://github.com/author/myproject/issues/42`, so using the query language is not a must. So in case we go with the `metadata:github-url:"https://github.com/author/myproject/issues/42"`, that would be fine: it would be a good syntax (later) for the webui, and the cmdline can use `metadata:github-url:\"https://github.com/author/myproject/issues/42\"` if it really wants to use the query language for metadata.

Is this OK? If so, then I could have a go at updating the code to do this.